projects/kit/components/input-phone/input-phone.component.ts

```diff
@@ -278,6 +259,32 @@ export class TuiInputPhoneComponent
         return !!this.search && isText(this.search);
     }
 
+    @tuiPure
+    private calculateMask(
+        countryCode: string,
+        phoneMaskAfterCountryCode: string,
+        nonRemovablePrefix: string,
+        allowText: boolean,
+    ): MaskitoOptions {
+        const mask = tuiCreatePhoneMaskExpression(countryCode, phoneMaskAfterCountryCode);
+
+        return allowText
+            ? {
+                  mask: ({value}) =>
+                      isText(value) && value !== '+'
+                          ? (MASKITO_DEFAULT_OPTIONS.mask as RegExp)
+                          : mask,
+              }
+            : {
+                  mask,
+                  preprocessor: tuiCreateCompletePhoneInsertionPreprocessor(
```

Comment:
We do not need those in the `allowText` version?

Reply:
fixed